R/dashboard.R

```diff
@@ -0,0 +1,119 @@
+#' R Markdown format for model monitoring dashboards
+#'
+#' @inheritParams pins::pin_write
+#' @param pins A list containing `board`, `name`, and `version`, as in
+#' [pins::pin_read()]
+#' @param display_pins Should the dashboard display a link to the pin(s)?
+#' Defaults to `TRUE`.
+#' @param ... Arguments passed to [flexdashboard::flex_dashboard()]
+#'
+#' @details The `vetiver_dashboard()` function is a specialized type of
+#' \pkg{flexdashboard}. See the flexdashboard website for additional
+#' documentation:
+#'  \href{http://rmarkdown.rstudio.com/flexdashboard/}{http://rmarkdown.rstudio.com/flexdashboard/}
+#'
+#' Before knitting the example `vetiver_dashboard()` template, execute the
+#' helper function `pin_example_kc_housing_model()` to set up demonstration
+#' model and metric pins needed for the monitoring demo. This function will:
+#'
+#' - fit an example model to training data
+#' - pin the vetiver model to your own [pins::board_local()]
+#' - compute metrics from testing data
+#' - pin these metrics to the same local board
+#'
+#' These are the steps you need to complete before setting up monitoring your
+#' real model.
+#'
+#' @export
+vetiver_dashboard <- function(pins, display_pins = TRUE, ...) {
+
+    rlang::check_installed("flexdashboard")
+    dashboard_dots <- rlang::list2(...)
+
+    if (display_pins) {
+        v <- pin_read_version(pins$board, pins$name, pins$version)
```

Comment:
This will indeed failed if the pins does not exist yet. (like when `pin_example_kc_housing_model()` has not be run prior to knit the document). 

If you want this format to run in example mode all the time, you coud try adding a `demo = TRUE` argument, and initialize the board in a temp board or local board before reading. But maybe that is too much...